src/components/Chart/plugins/chart.triggerOrigin.js

```diff
@@ -34,54 +34,47 @@
  * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import React from 'react';
-import { render, screen } from '../../utils/testUtils';
-import TimeSpanTop from '../Chart/TimeSpan/TimeSpanTop';
+/* eslint no-param-reassign: off */
 
-const initialState = {
-    app: {
-        chart: {
-            windowDuration: 1000,
-        },
-        trigger: {
-            triggerRunning: false,
-            triggerWindowOffset: 0,
-        },
-        app: {
-            capabilities: {
-                prePostTriggering: false,
-            },
-        },
-    },
-};
+import colors from '../../colors.scss';
+import { indexToTimestamp } from '../../../globals';
+
+const { gray700: color } = colors;
+
+const plugin = {
+    id: 'triggerorigin',
 
-test('should hide offset handler if device does not support it', () => {
-    render(<TimeSpanTop />, { initialState });
+    afterDraw(chartInstance) {
+        const {
+            chartArea: { top, bottom },
+            scales: { xScale },
+            chart: { ctx },
+            options: { triggerOrigin },
+        } = chartInstance;
+        if (!triggerOrigin) return;
+        // Not quite sure why we have to use the sample before, e.g. triggerOrigin - 1
```

Comment:
:)  when the buffer is initialised and empty (filled with undefined) the initial index is 0. When first element arrived, it is put in the buffer and the index is incremented, so the index points to the next free element. Depending where you set this index from, you may need to compensate for it.

Reply:
Thanks for clarifying!